spec/policy/oauth_mtls/oauth_mtls_spec.lua

```diff
@@ -0,0 +1,92 @@
+local _M = require('apicast.policy.oauth_mtls')
+local client = assert(fixture('CA', 'client.crt'))
+local X509 = require('resty.openssl.x509')
+local b64 = require('ngx.base64')
+
+local context = {}
+
+describe('oauth_mtls policy', function()
+  describe('.new', function()
+    it('works without configuration', function()
+      assert(_M.new())
+    end)
+  end)
+
+  describe('.access', function()
+    before_each(function()
+      context = {
+        jwt = {},
+        service = {
+          id = 4,
+          auth_failed_status = 403,
+          error_auth_failed = 'auth failed',
+          auth_failed_headers = 'text/plain; charset=utf-8'
+        }
+      }
+
+      ngx.header = {}
+      stub(ngx, 'print')
+      stub(ngx, 'exit')
+      context.jwt = {}
+    end)
+
+    it('accepts when the digest equals cnf claim', function()
+      ngx.var = { ssl_client_raw_cert = client }
+
+      context.jwt.cnf = {}

```

Comment:
what about a helper function: 

```
local jwt_cnf()
return { 'x5t#S256' = b64.encode_base64url(X509.parse_pem_cert(client):digest('SHA256')) }
```

and use something `context.jwt = jwt_cnf()`

Reply:
Changed.